### PR TITLE
Sentinel 1 Stats UDP ported to APEX

### PR DESCRIPTION
**Pull request by Pratichhya**
A User Defined Process (UDP) has been created for the Sentinel 1 Stats workflow. Ref: https://github.com/Open-EO/openeo-community-examples/blob/main/python/Sentinel1_Stats/Sentinel1_Stats.ipynb 

Included is a generate.py file that generates a process graph and saves it as a .json file. However, this process is not yet published or accessible as a shareable service. Within this file, I adopted the same method for establishing connection as for others

In the algorithm catalog JSON, a link to an OpenEO test result has been provided as a shareable link, for example. 

_Please note that this link may need to be updated._

### CODE REVIEW

**Comment by Pratichhya** [comment]
also could you please help me understand the cause of final action failure?

File: openeo_udp/sentinel1_stats/generate.py
Comment (jdries) [inline openeo_udp/sentinel1_stats/generate.py:17]: use spatial_extent
```suggestion
    spatial_extent = Parameter.spatial_extent( default={"west": 8.82, "south": 44.40, "east": 8.92, "north": 44.45})
```
(hope that suggestion works...)

File: openeo_udp/sentinel1_stats/generate.py
Comment (jdries) [inline openeo_udp/sentinel1_stats/generate.py:30]: bands could be a parameter, with default.

**Review by jdries — CHANGES_REQUESTED** [review]

File: algorithm_catalog/sentinel1_stats.json
Comment (HansVRP) [reply algorithm_catalog/sentinel1_stats.json:103]: _Documentation:  The most common way to do this is to store
it in a public git repository. Tagging a release is a good way to ensure that the link remains stable._

@jdries Is there already a dedicated repo or storage we should use? I want to avoid making use of VITO specific storage, as it will confuse external users

File: openeo_udp/sentinel1_stats/generate.py
Comment (jdries) [reply openeo_udp/sentinel1_stats/generate.py:30]: indeed, I don't think we can do dynamic band name generation yet, would need to dive into openeo processes to see if it is at all feasible with the current set.
So suggest to leave it to default bands for now.

File: algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md
Comment (soxofaan) [inline algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md:3]: some typo's here I think

```suggestion
This algorithm derives statistical insights from Sentinel-1 GRD data for a selected area and temporal extent. 
```

File: algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md
Comment (soxofaan) [inline algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md:6]: Also some typo's here

Overall there is quite some redundancy between the three sentences of this paragraph, they all say basically "we extract stats from EO data"

File: algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md
Comment (soxofaan) [inline algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md:10]: Python is irrelevant here I think: the process graph is fully defined as openEO JSON, without any trace of Python

File: algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md
Comment (soxofaan) [inline algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md:11]: This sentence just describes the algorithm, but has nothing to do with performance I think

File: algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md
Comment (soxofaan) [reply algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md:6]: Also, shouldn't the description clarify that this is about *temporal* statistics (stats along time dimension)?

File: algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md
Comment (soxofaan) [inline algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md:21]: I think it's confusing that you show only VH (I think) in the raw data, while showing VH and VV in the "mean" plot.
Also note that the labels on the y axis are different, making the comparison even more confusing

File: algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md
Comment (soxofaan) [inline algorithm_catalog/sentinel1_stats/sentinel1_stats_description.md:25]: I'm not sure this "larger extent" example adds a lot of value.


I think what would be more interesting is to show something else than "mean" as statistic

**Comment by HansVRP** [comment]
@jdries  The only failing unit tests seems to be related to the process graph not being on the actual repo, which should be resolved after the merge?

**Comment by soxofaan** [comment]
> also could you please help me understand the cause of final action failure?


from https://github.com/ESA-APEx/apex_algorithms/actions/runs/11323766302/job/31487115522?pr=36


```
=========================== short test summary info ============================
FAILED tests/test_scenarios.py::test_lint_scenario[sentinel1_stats] - requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://raw.githubusercontent.com/ESA-APEX/apex_algorithms/main/openeo_udp/sentinel1_stats.json
```

In your benchmark definition you use URL https://raw.githubusercontent.com/ESA-APEX/apex_algorithms/main/openeo_udp/sentinel1_stats.json  which does not exist yet as this PR is not merged yet.

This raises a bit a problem with this PR based approach at submitting algorithms: in the benchmark definition you have to point to the URL of the UDP, but what should you use? The URL of the feature branch being used to create the PR, or the URL of where the UDP will finally live after merging the PR (main branch)?

Two quick solutions:
- First make a PR of the UDP and algorithm definition and get that merged. Do the benchmark with a different PR
- Do a first PR with UDP, algorithm and benchmark definition, where you point to the UDP with the feature branch based URL. When that is merged, optionally: follow-up PR to update  URLs

**Comment by HansVRP** [comment]
@jdries @soxofaan I guess its ready for merge

**Review by soxofaan — APPROVED** [review]
yes I think it's fine to merge now

**Comment by jdries** [comment]
Ok, pushed the button, great to have another algorithm!

**Pull request merged**
